inspirehep/celery.py

```diff
@@ -25,15 +25,34 @@
 from __future__ import absolute_import, division, print_function
 
 import logging
+import os
+import signal
 
-from flask_celeryext import create_celery_app
+from flask import current_app
+from flask_celeryext import AppContextTask, create_celery_app
+from psycopg2 import OperationalError as Psycopg2OperationalError
+from sqlalchemy.exc import InvalidRequestError, OperationalError
 
-from .factory import create_app
+from inspirehep.factory import create_app
 
 
-celery = create_celery_app(
-    create_app(LOGGING_SENTRY_CELERY=True)
-)
+LOGGER = logging.getLogger(__name__)
+
+
+class CeleryTask(AppContextTask):
+
+    def on_failure(self, exc, task_id, args, kwargs, einfo):
+        if isinstance(exc, (InvalidRequestError, OperationalError, Psycopg2OperationalError)):
+            LOGGER.exception('Shutting down celery process because of'.format(exc))
+            path = current_app.config.get('TERMINATION_LOG_PATH', '/dev/termination-log')
+            with open(path, 'w') as term_log:
+                term_log.write(str(exc))
+
+            os.kill(os.getppid(), signal.SIGTERM)
```

Comment:
as discussed IRL, probably better not to use the config (no need to make this configurable, plus not sure we're in an app context here), and simply 
```python
        try:
            with open('/dev/termination-log', 'w') as term_log:
                term_log.write(str(exc))
        finally:
            os.kill(os.getppid(), signal.SIGTERM)
```